### PR TITLE
[Logging] Switch worker_setup_hook to worker_process_setup_hook

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Change worker_setup_hook -> worker_process_setup_hook

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

File: doc/source/ray-observability/user-guides/configure-logging.md [hunk 1]
```diff
@@ -384,10 +384,14 @@ ray.get(f.remote("A log message for a task."))
 
 ```{admonition} Caution
 :class: caution
-This is an experimental feature. It doesn't support [Ray Client](ray-client-ref) yet.
+This is an experimental feature. The semantic of the API is subject to change.
+It doesn't support [Ray Client](ray-client-ref) yet.
+
+When you use `ray.init(runtime_env={...})` with a [Ray Job Submission](jobs-quickstart) API (`ray job subimt --working-dir` or `ray job submit --runtime-env ...`), runtime_env passed to `ray.init` is ignored.
```
Comment (angelinalg) [inline doc/source/ray-observability/user-guides/configure-logging.md:390]:
```suggestion
When you use `ray.init(runtime_env={...})` with a [Ray Job Submission](jobs-quickstart) API (`ray job submit --working-dir` or `ray job submit --runtime-env ...`), runtime_env passed to `ray.init` is ignored.
```

Comment (angelinalg) [reply doc/source/ray-observability/user-guides/configure-logging.md:390]: This is hard to parse. Are you saying that these two switches don't work when you just Ray Jobs?

File: python/ray/_private/runtime_env/setup_hook.py [hunk 2]
```diff
@@ -27,15 +29,15 @@ def _encode_function_key(key: str) -> bytes:
...
     worker: "ray.Worker",
 ) -> Union[Dict[str, Any], RuntimeEnv]:
-    """Uploads the worker_setup_hook to GCS with a key.
+    """Uploads the worker_process_setup_hook to GCS with a key.
 
-    runtime_env["worker_setup_hook"] is converted to a decoded key
+    runtime_env["worker_process_setup_hook"] is converted to a decoded key
     that can load the worker setup hook function from GCS.
-    I.e., you can use internalKV.Get(runtime_env["worker_setup_hook])
+    I.e., you can use internalKV.Get(runtime_env["worker_process_setup_hook])
```
Comment (angelinalg) [inline python/ray/_private/runtime_env/setup_hook.py:40]:
```suggestion
    i.e., you can use internalKV.Get(runtime_env["worker_process_setup_hook])
```

File: python/ray/runtime_env/runtime_env.py [hunk 3]
```diff
@@ -235,8 +235,8 @@ class MyClass:
             The `run_options` list spec is here:
             https://docs.docker.com/engine/reference/run/
         env_vars: Environment variables to set.
-        worker_setup_hook: The setup hook that's called after workers
-            start and before tasks and actors are scheduled.
+        worker_process_setup_hook: (Experimental) The setup hook that's
+            called after workers start and before tasks and actors are scheduled.
             The value has to be a callable when passed to the job/task/actor.
```
Comment (angelinalg) [inline python/ray/runtime_env/runtime_env.py:240]:
```suggestion
            The value has to be a callable when passed to the Job, Task, or Actor.
```

File: python/ray/runtime_env/runtime_env.py [hunk 4]
```diff
@@ -235,8 +235,8 @@ class MyClass:
             The `run_options` list spec is here:
             https://docs.docker.com/engine/reference/run/
         env_vars: Environment variables to set.
-        worker_setup_hook: The setup hook that's called after workers
-            start and before tasks and actors are scheduled.
+        worker_process_setup_hook: (Experimental) The setup hook that's
+            called after workers start and before tasks and actors are scheduled.
```
Comment (angelinalg) [inline python/ray/runtime_env/runtime_env.py:239]:
```suggestion
            called after workers start and before Tasks and Actors are scheduled.
```

File: python/ray/runtime_env/runtime_env.py [hunk 5]
```diff
@@ -235,8 +235,8 @@ class MyClass:
...
             The value has to be a callable when passed to the job/task/actor.
             The callable is then exported and this value is converted to
             the setup hook's function name for the observability purpose.
```
Comment (angelinalg) [inline python/ray/runtime_env/runtime_env.py:242]:
```suggestion
            the setup hook's function name for observability.
```

**Review by angelinalg — APPROVED** [review]

File: doc/source/ray-observability/user-guides/configure-logging.md (hunk 1 above)
Comment (rkooo567) [reply doc/source/ray-observability/user-guides/configure-logging.md:390]: ray job ... is a job API. When you use the job API, the driver runtime env is ignored.

Comment (rkooo567) [reply doc/source/ray-observability/user-guides/configure-logging.md:390]: Any recommended sentences?

Comment (rkooo567) [reply doc/source/ray-observability/user-guides/configure-logging.md]: I changed it. Can you take a look? If there's no response by today, I will just merge a PR!
